databuilder/databuilder/task/neo4j_staleness_removal_task.py

```diff
@@ -210,47 +220,63 @@ def _validate_staleness_pct(self,
 
     def _validate_node_staleness_pct(self) -> None:
         total_nodes_statement = textwrap.dedent("""
-        MATCH (n)
-        WITH DISTINCT labels(n) as node, count(*) as count
+        MATCH (target:{type})
+        {optional_where}{extra_condition}
+        WITH DISTINCT labels(target) as node, count(*) as count
         RETURN head(node) as type, count
         """)
 
         stale_nodes_statement = textwrap.dedent("""
-        MATCH (n)
-        WHERE {}
-        WITH DISTINCT labels(n) as node, count(*) as count
+        MATCH (target:{{type}})
+        WHERE {{extra_condition}} {{optional_and}} {}
+        WITH DISTINCT labels(target) as node, count(*) as count
         RETURN head(node) as type, count
         """)
 
         stale_nodes_statement = textwrap.dedent(self._decorate_staleness(stale_nodes_statement))
 
-        total_records = self._execute_cypher_query(statement=total_nodes_statement)
-        stale_records = self._execute_cypher_query(statement=stale_nodes_statement,
-                                                   param_dict={MARKER_VAR_NAME: self.marker})
-        self._validate_staleness_pct(total_records=total_records,
-                                     stale_records=stale_records,
-                                     types=self.target_nodes)
+        for t in self.target_nodes:
+            if isinstance(t, TargetWithCondition):
+                formatted_total_nodes_statement = total_nodes_statement.format(type=t.target_type, optional_where='WHERE ', extra_condition=t.condition)
+                formatted_stale_nodes_statement = stale_nodes_statement.format(type=t.target_type, extra_condition=t.condition, optional_and='AND')
+            else:
+                formatted_total_nodes_statement = total_nodes_statement.format(type=t, optional_where='', extra_condition='')
+                formatted_stale_nodes_statement = stale_nodes_statement.format(type=t, extra_condition='', optional_and='')
```

Comment:
also this i think can be simplified. Inside the `isinstance(t, TargetWithCondition)` just set the interesting variables and do the formatting once below.